Library/Homebrew/rubocops/uses_from_macos.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require "rubocops/extend/formula"
+
+module RuboCop
+  module Cop
+    module FormulaAuditStrict
+      # This cop audits `uses_from_macos` dependencies in formulae
+      class UsesFromMacos < FormulaCop
+        ALLOWED_USES_FROM_MACOS_DEPS = %w[
+          bison
+          bzip2
+          curl
+          expat
+          flex
+          libffi
+          libxml2
+          libxslt
+          ncurses
+          m4
+          perl
+          php
+          python@2
+          ruby
+          sqlite
+          texinfo
+          unzip
+          vim
+          xz
+          zlib
+        ].join("|")
+
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          find_method_with_args(body_node, :uses_from_macos, /^(#{ALLOWED_USES_FROM_MACOS_DEPS})/).each do |method|
```

Comment:
Turns out that this regex is broken. If I put the same thing in Rubular, what I want it to do is match things that aren't in `ALLOWED_USES_FROM_MACOS_DEPS`, but it does the opposite: https://rubular.com/r/hiIOJfrA3MlYFS. I might have to use `find_every_method_call_by_name` instead and do the allowed formulae filtering later.

Reply:
Does negative matching work? `/^(?!#{ALLOWED_USES_FROM_MACOS_DEPS}).*/`

Reply:
It doesn't seem to, unless I'm doing something wrong.

Reply:
Wait, yes it does. I _was_ doing something wrong. :roll_eyes: